test/test_objectid.py

```diff
@@ -166,414 +166,40 @@ def test_is_valid(self):
-    fnv_test_str[162]: 0x17b6b2,
-    fnv_test_str[163]: 0xad4cfb,
-    fnv_test_str[164]: 0x256811,
-    fnv_test_str[165]: 0xb18dd8,
-    fnv_test_str[166]: 0x61c153,
-    fnv_test_str[167]: 0x47d20d,
-    fnv_test_str[168]: 0x8b689f,
-    fnv_test_str[169]: 0xd2a40b,
-    fnv_test_str[170]: 0x549b0a,
-    fnv_test_str[171]: 0xe1b55b,
-    fnv_test_str[172]: 0x0cd3d1,
-    fnv_test_str[173]: 0x471605,
-    fnv_test_str[174]: 0x5eef10,
-    fnv_test_str[175]: 0xed3629,
-    fnv_test_str[176]: 0x624952,
-    fnv_test_str[177]: 0x9b8688,
-    fnv_test_str[178]: 0x15e25f,
-    fnv_test_str[179]: 0xa98d05,
-    fnv_test_str[180]: 0xdf8bcc,
-    fnv_test_str[181]: 0x1e9051,
-    fnv_test_str[182]: 0x3f70db,
-    fnv_test_str[183]: 0x95aedb,
-    fnv_test_str[184]: 0xa7f7d7,
-    fnv_test_str[185]: 0x3bc660,
-    fnv_test_str[186]: 0x610967,
-    fnv_test_str[187]: 0x157785,
-    fnv_test_str[188]: 0x2b2800,
-    fnv_test_str[189]: 0x8239ef,
-    fnv_test_str[190]: 0x5869f5,
-    fnv_test_str[191]: 0x415c76,
-    fnv_test_str[192]: 0xe4ff6f,
-    fnv_test_str[193]: 0xb7977d,
-    fnv_test_str[194]: 0xa43a7b,
-    fnv_test_str[195]: 0xb3be1e,
-    fnv_test_str[196]: 0x777aaf,
-    fnv_test_str[197]: 0x21c38a,
-    fnv_test_str[198]: 0x9d0839,
-    fnv_test_str[199]: 0x823d2f,
-    fnv_test_str[200]: 0xa27250,
-    fnv_test_str[201]: 0xc5c656,
-    fnv_test_str[202]: 0x3b0800,
-}
-
-class TestFNV1a24(unittest.TestCase):
-
-    def test_fnv_1a_24(self):
-        for key in fnv1a_24_vector:
-            self.assertEqual(_fnv_1a_24(key), fnv1a_24_vector[key])
+    def test_counter_overflow(self):
+        # Spec-test to check counter overflows from max value to 0.
+        ObjectId._inc = _MAX_COUNTER_VALUE
+        ObjectId()
+        self.assertEqual(ObjectId._inc, 0)
+
+    def test_timestamp_values(self):
+        # Spec-test to check timestamp field is interpreted correctly.
+        TEST_DATA = {
+            0x00000000: (1970, 1, 1, 0, 0, 0),
+            0x7FFFFFFF: (2038, 1, 19, 3, 14, 7),
+            0x80000000: (2038, 1, 19, 3, 14, 8),
+            0xFFFFFFFF: (2106, 2, 7, 6, 28, 15),
+        }
+
+        def generate_objectid_with_timestamp(timestamp):
+            oid = ObjectId()
+            _, trailing_bytes = struct.unpack(">IQ", oid.binary)
+            new_oid = struct.pack(">IQ", timestamp, trailing_bytes)
+            return ObjectId(new_oid)
+
+        for tstamp, exp_datetime_args in TEST_DATA.items():
+            oid = generate_objectid_with_timestamp(tstamp)
+            self.assertEqual(
+                oid.generation_time,
+                datetime.datetime(*exp_datetime_args, tzinfo=utc))
+
+    def test_random_regenerated_on_pid_change(self):
+        # Test that change of pid triggers new random number generation.
+        random_original = ObjectId._random()
+        ObjectId._pid += 1
+        random_new = ObjectId._random()
+        self.assertNotEqual(random_original, random_new)
+        ObjectId._pid -= 1
```

Comment:
This line isn't needed.

Reply:
I only put this in here so that I don't end up changing the 'perceived' PID (i.e. the PID tracked by the `ObjectId` class) of the test runner's context such that it results in failures in other unrelated tests that depend on the PID being same. Obviously, that is an extreme situation and I doubt anything would ever rely upon the consistency of ObjectId w.r.t PID across test classes so I'll remove it.

Reply:
It's not needed because the `_random` method sets _pid to the current PID. Subtracting at the makes the _pid different *again*.